services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/HanketunnusServiceImplTest.kt

```diff
@@ -0,0 +1,24 @@
+package fi.hel.haitaton.hanke
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.jupiter.api.Test
+import java.time.ZonedDateTime
+
+class HanketunnusServiceImplTest {
+
+    @Test
+    fun newHanketunnus() {
+        val repository: IdCounterRepository = mockk()
+        val service = HanketunnusServiceImpl(repository)
+
+        every { repository.incrementAndGet(CounterType.HANKETUNNUS.name) }.returns(listOf(IdCounter(CounterType.HANKETUNNUS, 1)))
+
+        val hankeTunnus = service.newHanketunnus()
+        val currentYear = ZonedDateTime.now(TZ_UTC).year
+
+        assertThat(hankeTunnus).isEqualTo("HAI${currentYear.toString().substring(2)}-1")
```

Comment:
For readability maybe add a comment about the hankeTunnus format we expect..? I use tests sometimes to explain how things should work... it might help others too

Reply:
Yes, you are right.